### ISSUE TITLE
quickstarts/excel-quickstart-vue certificate problem

### ISSUE THREAD

**Issue by LewisPringle** [OfficeDev/office-js-docs-pr#720]
I tried following your instructions. 

They read: "In a web browser, navigate to https://localhost:8080. If your browser indicates that the site's certificate is not trusted, you will need to configure your computer to trust the certificate."

When I run 
   curl -v https://localhost:8080/
I get:
curl: (77) schannel: next InitializeSecurityContext failed: SEC_E_UNTRUSTED_ROOT (0x80090325) - The certificate chain was issued by an authority that is not trusted.

when I access the website with internet explorer, npm crashes with:
Your application is running here: http://localhost:8080
buffer.js:72
class FastBuffer extends Uint8Array {}
^

RangeError: Invalid typed array length: -4095
    at new Uint8Array (&lt;anonymous&gt;)
    at new FastBuffer (buffer.js:72:1)
    at Handle.onStreamRead [as onread] (internal/stream_base_commons.js:144:17)
    at Stream.&lt;anonymous&gt; (C:\Sandbox\EY\office-addins-experiments\my-add-in\node_modules\handle-thing\lib\handle.js:120:12)
    at Stream.emit (events.js:193:15)
    at endReadableNT (C:\Sandbox\EY\office-addins-experiments\my-add-in\node_modules\readable-stream\lib\_stream_readable.js:1010:12)
    at process.internalTickCallback (internal/process/next_tick.js:72:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! my-add-in@1.0.0 dev: `webpack-dev-server --https --inline --progress --config build/webpack.dev.conf.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the my-add-in@1.0.0 dev script.
npm ERR! This is probably not a problem

- note - I'm using node v11.6.0.

Your docs say to fix the cert, but I have no idea how ttodo that or what cert to add?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ea230e95-f483-6d19-8167-2341a03cf396
* Version Independent ID: 3846765f-7587-fec4-711e-84edfce9c30c
* Content: [Build an Excel add-in using Vue - Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/excel-quickstart-vue)
* Content Source: [docs/quickstarts/excel-quickstart-vue.md](https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/quickstarts/excel-quickstart-vue.md)
* Product: **office 365**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by AlexJerabek** [comment]
Hi @LewisPringle, 

I'm sorry to hear you're having problems with the quick start. Here are the steps for [Adding Self-Signed Certificates as Trusted Root Certificate](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md). We seem to have links to that article in the [jQuery quick start](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/excel-quickstart-jquery?tabs=visual-studio-code#start-the-dev-server), but not the vue one.

Please let me know if that unblocks you. I'll update the vue quick start to include this information.

**Comment by AlexJerabek** [comment]
@LewisPringle, Ignore my previous advice. the certificates are generated differently in the vue quick start. I'm researching the correct guidance now.

**Comment by AlexJerabek** [comment]
@LewisPringle If you are using Windows, [these steps from an older guide](https://github.com/OfficeDev/generator-office/blob/fd600bbe00747e64aa5efb9846295a3f66d428aa/src/docs/ssl.md#add-certification-file-through-ie) should get the certificates trusted and unblock you.

Please let me know if that solves your problem. Thanks!

**Comment by kbrandl** [comment]
@LewisPringle thanks for the update; glad to hear that you're unblocked. I'm going to close this issue, as we've created an internal work item (2907215) to track the need to update the Vue quick start article with info about how to trust the cert.

**Comment by Rednas83** [comment]
There is a new fix. The old one is based on then npm package browser-sync.
https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md

Unfortunately it only works in IE and introduces a DOM7011 warning. Resulting in a page with INET_E_RESOURCE_NOT_FOUND (DNS error).
Note: In chrome/edge there is still the HTTP401 error. Also after following the previous fix.

Please re-open the issue.

**Comment by AlexJerabek** [comment]
Hi @Rednas83, 

I just went through the vue quickstart. If you're using Chrome, you'll have to copy the certificate to a file (see picture), then use the [Management Console to install the certificate](https://github.com/OfficeDev/generator-office/blob/fd600bbe00747e64aa5efb9846295a3f66d428aa/src/docs/ssl.md#add-certification-file-through-management-console). It'll have an ".cer" extension instead of ".crt", but it should work all the same. 

![image](https://user-images.githubusercontent.com/38896772/78924165-2da44700-7a4e-11ea-8e70-f66fc8ec78a9.png)

The browser-sync reference in the other article is misleading. I've opened an issue in the yo office repo to alert them to the problem: https://github.com/OfficeDev/generator-office/issues/524

I'd encourage you to reply to that issue and help raise its visibility. I'll also tag @nokafor for awareness.

Please let me know if that Chrome certificate installation information helps unblock you. I'll leave this issue closed, since the documentation error is in another repo and there is an issue tracking it there.

**Comment by Rednas83** [comment]
Thanx @AlexJerabek.  I am now unblocked too. Also don't forget to restart the dev server after importing the copied certifcate (.cer) into the management console (mmc).

**Comment by Rednas83** [comment]
Is anyone able to run the quickstart without the HTTP403/HTTP404 error?

**Comment by AlexJerabek** [comment]
@Rednas83 I'll reopen the issue and start investigating this. Tagging @akrantz for additional guidance on the latest issue in this thread.

**Comment by akrantz** [comment]
Adding @TCourtneyOwen to the thread. @AlexJerabek, it is difficult to understand what is the issue now from this re-opened thread. It would help if you can reproduce the issue and summarize what you're seeing.

**Comment by AlexJerabek** [comment]
Hi @Rednas83, 
I just re-ran the vue quick start and tested the add-in. I also checked accessing `https://localhost:3000` through both Edge and Chrome. Everything seems to be working fine. I did not encounter the 403/404 error.
Check you double check you're using the updated quick start (patched with https://github.com/OfficeDev/office-js-docs-pr/pull/1737)? Also, be sure that the `vue.config.js` file is in the root folder of the vue project, not the `src` folder.

@akrantz, Thanks for responding. The active issue is the 403/404 error @Rednas83 is reporting in their last couple posts.

**Comment by Rednas83** [comment]
After opening https://localhost:3000/ in Edge I get
![image](https://user-images.githubusercontent.com/25219229/80363781-01c5f580-8885-11ea-9930-3aef2a8b1b72.png)
After opening https://localhost:3000/taskpane.html in Edge I get the same
![image](https://user-images.githubusercontent.com/25219229/80364040-7e58d400-8885-11ea-8405-1ea44d4477c4.png)
After opening in Excel everything works
![image](https://user-images.githubusercontent.com/25219229/80364174-b9f39e00-8885-11ea-8a1e-5f453da262da.png)

Tried:
1. Installing certificate
![image](https://user-images.githubusercontent.com/25219229/80366389-7bf87900-8889-11ea-87d6-2fa655d13c2b.png)
2. Disabling AdBlock extension (only pausig AdBlock is not sufficient!)
![image](https://user-images.githubusercontent.com/25219229/80364687-9f6df480-8886-11ea-8a08-97dec7614a39.png)
3. Different example https://docs.microsoft.com/en-us/office/dev/add-ins/tutorials/excel-tutorial but is not loading the body
![image](https://user-images.githubusercontent.com/25219229/80365338-b2cd8f80-8887-11ea-950c-886f80687782.png)

Could it be my excel version?
![image](https://user-images.githubusercontent.com/25219229/80365698-5028c380-8888-11ea-8836-5df67fc74807.png)

Not sure what I can try more. Followed the guide a couple of times now and keep getting the same result.

**Comment by AlexJerabek** [comment]
@Rednas83, 

Thanks for the additional information. Based on your reporting, the issue is not just in the vue quickstart, but in the JQuery one as well, correct? 

I'm going to open a new issue tracking that problem, since it is possibility unrelated to this older issue.
